StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1017CodeFixProvider.cs

```diff
@@ -52,11 +53,17 @@ public override FixAllProvider GetFixAllProvider()
                 if (!precedingToken.TrailingTrivia.Any(SyntaxKind.WhitespaceTrivia))
                     continue;
 
-                SyntaxToken corrected = precedingToken.WithoutTrailingWhitespace().WithoutFormatting();
-                SyntaxNode transformed = root.ReplaceToken(precedingToken, corrected);
-                Document updatedDocument = context.Document.WithSyntaxRoot(transformed);
-                context.RegisterCodeFix(CodeAction.Create("Fix spacing", t => Task.FromResult(updatedDocument)), diagnostic);
+                context.RegisterCodeFix(CodeAction.Create("Fix spacing", t => GetTransformedDocument(context, root, precedingToken)), diagnostic);
```

Comment:
:exclamation: Move as much of the logic from this loop to the callback as possible.